tools/packaging/kata-deploy/Dockerfile

```diff
@@ -1,45 +1,23 @@
-# Copyright Intel Corporation.
+# Copyright Intel Corporation, 2022 IBM Corp.
 #
 # SPDX-License-Identifier: Apache-2.0
 
-FROM registry.centos.org/centos:7 AS base
-
-ENV container docker
-
-RUN (cd /lib/systemd/system/sysinit.target.wants/ && for i in *; do [ "$i" = systemd-tmpfiles-setup.service ] || rm -f "$i"; done); \
-rm -f /lib/systemd/system/multi-user.target.wants/*; \
-rm -f /etc/systemd/system/*.wants/*; \
-rm -f /lib/systemd/system/local-fs.target.wants/*; \
-rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
-rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
-rm -f /lib/systemd/system/basic.target.wants/*; \
-rm -f /lib/systemd/system/anaconda.target.wants/*;
```

Comment:
You cannot remove this bits, @Jakob-Naucke, as those are needed to have an image that's systemd capable.
Why do we need that, you may ask? Because of operations like: https://github.com/kata-containers/kata-containers/blob/7db9bef72ce2f99a2924fa6c7fd4f9706ded5305/tools/packaging/kata-deploy/scripts/kata-deploy.sh#L67-L68

I'd be fine switching any other distro, preferably smaller than CentOS, preferably with multi arch support, as long as we don't lose functionality.

If you're solving this in a different way and I missed it, please, just let me know.

Reply:
The image must be systemd-capable, but it is not systemd capable because we remove some systemd targets, but because of mounts. kata-deploy works without these removals.

Reply:
> I'd be fine switching any other distro, preferably smaller than CentOS, preferably with multi arch support, as long as we don't lose functionality.

We're currently stuck with systemd <= ~237 (see #3723), and I think CentOS/ClefOS 7 are the only non-commercial distros that still support this :/

Reply:
> The image must be systemd-capable, but it is not systemd capable because we remove some systemd targets, but because of mounts. kata-deploy works without these removals.

Yep, I should have been more explicit marking every part of the file that worried me, such as the volume removal as well.

Anyways, test-kata-deploy seems to be happy enough about this, so I am fine with the changes.